lib/type-helpers/intersection-type.helper.ts

```diff
@@ -0,0 +1,44 @@
+import { Type } from '@nestjs/common';
+import { Field } from '../decorators';
+import { getFieldsAndDecoratorForType } from '../schema-builder/utils/get-fields-and-decorator.util';
+import {
+  ClassDecoratorFactory,
+  inheritValidationMetadata,
+  inheritTransformationMetadata,
+} from './type-helpers.utils';
+
+export function IntersectionType<A, B>(
+  classARef: Type<A>,
+  classBRef: Type<B>,
+  decorator?: ClassDecoratorFactory,
+): Type<A & B> {
+  const forA = getFieldsAndDecoratorForType(classARef);
+  const forB = getFieldsAndDecoratorForType(classBRef);
+  const decoratorFactory = forA.decoratorFactory;
+  const fields = [...forA.fields, ...forB.fields];
+
+  abstract class IntersectionObjectType {}
+  decoratorFactory({ isAbstract: true })(IntersectionObjectType);
+  if (decorator) {
+    decorator({ isAbstract: true })(IntersectionObjectType);
+  } else {
+    decoratorFactory({ isAbstract: true })(IntersectionObjectType);
+  }
+
+  inheritValidationMetadata(classARef, IntersectionObjectType);
+  inheritTransformationMetadata(classARef, IntersectionObjectType);
+  inheritValidationMetadata(classBRef, IntersectionObjectType);
+  inheritTransformationMetadata(classBRef, IntersectionObjectType);
+
+  fields.forEach((item) => {
+    Field(item.typeFn, { ...item.options })(
+      IntersectionObjectType.prototype,
+      item.name,
+    );
+  });
+
+  Object.defineProperty(IntersectionObjectType, 'name', {
+    value: `Mixin${classARef.name}${classBRef.name}`,
```

Comment:
Oh, I forgot to rename this one. Is that OK? 

And there is a question about the length of `name`. If someone nested using the helpers, it could be very long. Will this cause problems? @kamilmysliwiec

Reply:
Actually, I think you can remove the `Object.defineProperty` call. :)

Reply:
Previously, when dealing with dynamic decorated class generation, if the classname hadn't modified, it won't be working. 😕 

https://github.com/nestjsx/crud/pull/25/files

Reply:
Oh, so yeah, there's a risk that someone would use the returned class directly (without extending it by another class). Let's change `Mixin` to `Intersection` then and leave the `Object.defineProperty` call (just in case)